src/screens/Profile/Notification.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { View, StyleSheet, AsyncStorage } from 'react-native';
+import { Card, Button, Paragraph } from 'react-native-paper';
+import auth from '@react-native-firebase/auth';
+import Swipeable from 'react-native-gesture-handler/Swipeable';
+import PropTypes from 'prop-types';
+
+export default class Notification extends React.Component {
+  constructor(props) {
+    super(props);
+    const { viewed } = this.props;
+    this.state = { viewed, visible: true };
+    this.onClearPress = this.onClearPress.bind(this);
+    this.onViewPress = this.onViewPress.bind(this);
+  }
+
+  componentDidMount() {
+    this.unsubscribeFromAuth = auth().onAuthStateChanged((user) => {
+      if (user) {
+        this.setState({ currentUserID: user.uid });
+      }
+    });
+  }
+
+  async onViewPress() {
+    const { notificationID } = this.props;
+    const { currentUserID } = this.state;
+    await AsyncStorage.getItem(currentUserID).then((value) => {
+      const notificationArray = JSON.parse(value);
+      const notif = notificationArray.find((el) => el.id === notificationID);
+      if (notif.viewed === false) {
+        const updatedArray = notificationArray.filter((val) => val.id !== notificationID);
+        notif.viewed = true;
+        updatedArray.push(notif);
+        AsyncStorage.setItem(currentUserID, JSON.stringify(updatedArray));
+        this.setState({ viewed: true });
+      }
+    });
+  }
+
+  async onClearPress() {
+    const { notificationID } = this.props;
+    const { currentUserID } = this.state;
+    await AsyncStorage.getItem(currentUserID).then(async (value) => {
+      if (value) {
+        const notificationArray = JSON.parse(value);
+        const updatedArray = notificationArray.filter((val) => val.id !== notificationID);
+        await AsyncStorage.setItem(currentUserID, JSON.stringify(updatedArray));
+        this.setState({ visible: false });
+      }
+    });
```

Comment:
Same as above

Reply:
For me: Utilized filter in conjunction with push because I wanted the viewed notification to be in the end of the notifications array, so that when our component was re-rendered, viewed notifications show up last among our notifications

Reply:
Ok that should be fine!